cmd/cronosd/cmd/fix-unlucky-tx_test.go

```diff
@@ -105,15 +105,15 @@ func TestFindUnluckyTx(t *testing.T) {
 			for _, txResults := range tc.txResults {
 				txResult := txResults[0]
 				blockRes.DeliverTxs = append(blockRes.DeliverTxs, &txResult.Result)
-				if len(txResults) > 1 {
+				hasIndexedTxResultToOverwrite := len(txResults) > 1
+				if hasIndexedTxResultToOverwrite {
 					txResult = txResults[1]
 				}
 				if txResult != nil {
```

Comment:
`txResult` is either `txResults[0]` or `txResults[1]`, it's never `nil`?

Reply:
yup, in this test

Reply:
Still don't understand the code logic here, you append `txResults[0]` to `blockRes.DeliverTxs`, and index `txResults[1]`, append either `txResults[0]` or `txResults[1]` to `block.Txs`, what does that mean?

Reply:
for case [3](https://github.com/mmsqe/cronos/blob/3ef29ec5574891e312390f75489cf1e18d2479e3/cmd/cronosd/cmd/fix-unlucky-tx_test.go#L83), `mockResult(encCfg.TxConfig, 1, false)` is txResult with code 11 to test [exceed](https://github.com/mmsqe/cronos/blob/3ef29ec5574891e312390f75489cf1e18d2479e3/cmd/cronosd/cmd/fix-unlucky-tx.go#L317) case, but modify [indexer](https://github.com/mmsqe/cronos/blob/3ef29ec5574891e312390f75489cf1e18d2479e3/cmd/cronosd/cmd/fix-unlucky-tx.go#L320) with success `mockResult(encCfg.TxConfig, 1, true)` code

Reply:
Do you mean sth like this:
```
type MockTxResut {
  Result *abci.TxResult
  ReplayedResult *abci.TxResult
}
```

Reply:
But after replay `blockRes.DeliverTxs` should be updated too?

Reply:
If DeliverTxs update together with block.Txs and txIndexer, will we case overwriten bug

Reply:
> If `DeliverTxs` update together with `block.Txs` and txIndexer, will we have overwritten bug

what do you mean?

Reply:
when we replay and patch a tx, we'll update the tx indexer and block results together.